svc-aggregation/system/rediscoverinventory.go

```diff
@@ -132,11 +135,16 @@ func (e *ExternalInterface) RediscoverSystemInventory(deviceUUID, systemURL stri
 	h.TraversedLinks = make(map[string]bool)
 	progress := int32(100)
 	systemsEstimatedWork := int32(75)
-	_, progress, _ = h.getSystemInfo("", progress, systemsEstimatedWork, req)
-	//rediscovering the Chassis Information
-	req.OID = "/redfish/v1/Chassis"
-	chassisEstimatedWork := int32(15)
-	progress = h.getAllRootInfo("", progress, chassisEstimatedWork, req)
+	if strings.Contains(systemURL, "/Storage") {
+		_, progress, _ = h.getStorageInfo("", progress, systemsEstimatedWork, req)
```

Comment:
If task is being used in your function y send empty string? define your function without task

Reply:
Incorporated review comments